packages/malloy/src/model/malloy_query.ts

```diff
@@ -2269,9 +2269,9 @@ class QueryQuery extends QueryField {
       resultMetadata: this.getResultMetadata(this.rootResult),
       type: 'struct',
     };
-    if (isQuerySegment(this.firstSegment) && this.firstSegment.queryTimezone) {
-      outputStruct.queryTimezone = this.firstSegment.queryTimezone;
-    }
+
+    outputStruct.queryTimezone = resultStruct.getQueryInfo().queryTimezone;
```

Comment:
@mtoy-googly-moogly I'd really like to get your feedback on these lines in particular, this change enabled the data to flow correctly, jsut want to confirm that I'm not doing anything crazy.

Reply:
Done. Those changes went in with parent PR but applied your suggestion here.